contractsV5/eth2dai/Eth2DaiReserve.sol

```diff
@@ -0,0 +1,846 @@
+
+        if (tokenVal <= MAX_QTY) { return (shouldUse, premiumBps); }
+
+        InternalInventoryData memory inventoryData = getInternalInventoryData(token);
+        premiumBps = inventoryData.premiumBps;
+
+        uint tokenBalance = token.balanceOf(address(this));
+
+        if (ethToToken) {
+            if (tokenBalance < tokenVal) { return (shouldUse, premiumBps); }
+            if (tokenBalance - tokenVal < inventoryData.minTokenBal) { return (shouldUse, premiumBps); }
+        } else {
+            if (address(this).balance < ethVal) { return (shouldUse, premiumBps); }
+            if (tokenBalance + tokenVal > inventoryData.maxTokenBal) { return (shouldUse, premiumBps); }
+        }
+
+        if (!checkValidSpread(bid, ask, true, inventoryData.minSpreadBps)) {
+            return (shouldUse, premiumBps);
+        }
+
+        shouldUse = true;
+    }
+
+    function applyInternalInventoryHintToRate(
+        uint rate,
+        bool useInternalInventory
+    )
+        internal pure
+        returns(uint)
+    {
+        return rate % 2 == (useInternalInventory ? 1 : 0)
+            ? rate
+            : rate - 1;
+    }
+
+    function valueAfterReducingFee(uint val) internal view returns(uint) {
+        require(val <= MAX_QTY, "valueAfterReducingFee: val > MAX_QTY");
+        return ((10000 - feeBps) * val) / 10000;
+    }
+
+    function valueAfterAddingPremium(uint val, uint premium) internal pure returns(uint) {
+        require(val <= MAX_QTY, "valueAfterAddingPremium: val > MAX_QTY");
+        return val * (10000 + premium) / 10000;
+    }
+
+    function findBestOffers(
+        ERC20 dstToken,
+        ERC20 srcToken,
+        uint srcAmount,
+        OfferData memory bid,
+        OfferData memory ask
+    )
+        internal view
+        returns(uint totalDestAmount, OfferData[] memory offers)
+    {
+        uint remainingSrcAmount = srcAmount;
+        uint maxOrdersToTake;
+        uint maxTraversedOrders;
+        uint minPayAmount;
+        uint numTakenOffer = 0;
+        totalDestAmount = 0;
+        ERC20 token = srcToken == wethToken ? dstToken : srcToken;
+
+        (maxOrdersToTake, maxTraversedOrders, minPayAmount) = calcOfferLimitsFromFactorData(
+            token,
+            (srcToken == wethToken),
+            bid,
+            ask,
+            srcAmount
+        );
+
+        offers = new OfferData[](maxTraversedOrders);
+
+        // return earlier, we don't want to take any orders
+        if (maxTraversedOrders == 0 || maxOrdersToTake == 0) {
+            return (totalDestAmount, offers);
+        }
+
+        // otc's terminology is of offer maker, so their sellGem is our (the taker's) dest token.
+        // if we don't have best offers, get them.
+        if ((srcToken == wethToken && bid.id == 0) || (dstToken == wethToken && ask.id == 0)) {
+            offers[0].id = otc.getBestOffer(dstToken, srcToken);
```

Comment:
can we reach here with bid.id == 0 or ask.id == 0?

Reply:
Yes, we still can reach here if there are not offers